ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java

```diff
@@ -137,42 +139,30 @@ public void testDataStream(){
   public void runTestDataStream(){
     final int bufferSize = 1024*1024;
     final int bufferNum = 10;
-    final DataStreamOutput out = client.stream();
-
-    //send request
-    final List<CompletableFuture<DataStreamReply>> futures = new ArrayList<>();
-    futures.add(sendRequest(out, 1024));
-
-    //send data
-    final int halfBufferSize = bufferSize/2;
-    int dataSize = 0;
-    for(int i = 0; i < bufferNum; i++) {
-      final int size = halfBufferSize + ThreadLocalRandom.current().nextInt(halfBufferSize);
-      final ByteBuffer bf = initBuffer(dataSize, size);
-      futures.add(out.streamAsync(bf));
-      dataSize += size;
-    }
 
-    //join all requests
-    for(CompletableFuture<DataStreamReply> f : futures) {
-      f.join();
-    }
-    Assert.assertEquals(dataSize, byteWritten);
-    shutDownSetup();
-  }
+    ByteBuffer buf = initBuffer(bufferSize);
+    Message msg = Message.valueOf(ByteString.copyFrom(buf));
+    SizeInBytes size = SizeInBytes.valueOf(bufferSize / bufferNum);
+    DataStreamOutput out = client.stream();
+    CompletableFuture<DataStreamReply> f = client.streamAsync(msg, size, out);
+    f.join();
+
+    DataStreamClientImpl.DataStreamOutputImpl impl = (DataStreamClientImpl.DataStreamOutputImpl) out;
+    Assert.assertEquals(writeRequest.getClientId(), impl.getRequest().getClientId());
+    Assert.assertEquals(writeRequest.getCallId(), impl.getRequest().getCallId());
+    Assert.assertEquals(writeRequest.getRaftGroupId(), impl.getRequest().getRaftGroupId());
+    Assert.assertEquals(writeRequest.getServerId(), impl.getRequest().getServerId());
 
-  CompletableFuture<DataStreamReply> sendRequest(DataStreamOutput out, int size) {
-    // TODO RATIS-1085: create a RaftClientRequest and put it in the buffer
```

Comment:
The idea for this TODO is to create a RaftClientRequest without a Message and then send it out as the header.  The the data is sent following the request using ByteBuffer.  We won't create Message here in order to support zero buffer copying.